invenio_query_parser/utils.py

```diff
@@ -68,17 +68,33 @@ def generate_keywords():
         from invenio_base.globals import cfg
         keyword_mapping = cfg['SEARCH_ELASTIC_KEYWORD_MAPPING']
         json_schema_paths = cfg['JSON_SCHEMA_PATHS']
+        elastic_config_paths = cfg['ELASTIC_MAPPINGS_PATHS']
     except (ImportError, RuntimeError):
         from .config import DEFAULT_KEYWORDS as keyword_mapping
         json_schema_paths = []
+        elastic_config_paths = []
+    # Get keywords from configuration file
     keywords = keyword_mapping.keys()
     for k in keyword_mapping.values():
         if isinstance(k, dict):
             keywords += k.keys()
+    # Get keywords from the json schema
     for path in json_schema_paths:
         with open(path) as data_file:
             data = json.load(data_file)
         data = data.get('properties')
         dotted_keywords = get_dotted_keys(data, '', [])
         keywords += dotted_keywords
-    return keywords
+    # Get keywords from elasticsearch mapping
+    for path in elastic_config_paths:
+        with open(path) as data_file:
+            data = json.load(data_file)
+        data = data.get('mappings').get('record').get('properties')
+        dotted_keywords = get_dotted_keys(data, '', [])
+        keywords += dotted_keywords
+    # Remove duplicates.
+    keyword_list = []
+    for i in keywords:
+        if i not in keyword_list:
+            keyword_list.append(i)
```

Comment:
You can just define `keyword` as a `set` and in this way you will not have duplicates :smile: